### PR TITLE
Forbidden filesystem opcodes from accessing outside game locations.

### PR DESCRIPTION
**Pull request by MiranDMC**
image

### CODE REVIEW

File: cleo_plugins/FileSystemOperations/FileSystemOperations.cpp [hunk 1]
```diff
@@ -114,6 +118,7 @@ class FileSystemOperations
     static OpcodeResult WINAPI opcode_0A9A(CRunningScript* thread)
     {
         OPCODE_READ_PARAM_FILEPATH(filename);
+        PATH_SECURITY_CHECK(filename);
```
Comment (x87) [inline cleo_plugins/FileSystemOperations/FileSystemOperations.cpp:121]: why not make it part of OPCODE_READ_PARAM_FILEPATH?

Comment (MiranDMC) [reply cleo_plugins/FileSystemOperations/FileSystemOperations.cpp:121]: As not all file path operations need check like that.
For example set work directory. It does not access that path in any way but will be later used with relative paths.

Comment (x87) [reply cleo_plugins/FileSystemOperations/FileSystemOperations.cpp:121]: we should not allow setting work directory outside of game directory or game settings either

Comment (MiranDMC) [reply cleo_plugins/FileSystemOperations/FileSystemOperations.cpp:121]: I do not know, but just act like that is not forbidden as it not accesses the location in any way. Later you might use relative paths like "Program Files\Gta\..."

Comment (x87) [reply cleo_plugins/FileSystemOperations/FileSystemOperations.cpp:121]: I don't see why a script would need to work with outside folders. But anyway we can restrict all filepaths by default and make exceptions on case by case basis for certain commands.

**Review by x87 — APPROVED** [review]

File: cleo_plugins/FileSystemOperations/FileSystemOperations.cpp (hunk 1 above)
Comment (MiranDMC) [reply cleo_plugins/FileSystemOperations/FileSystemOperations.cpp]: Reworked

**Review by x87 — APPROVED** [review]

**Pull request merged**